torch/csrc/jit/passes/quantization.cpp

```diff
@@ -131,43 +143,82 @@ static void addObserverFor(Value* v, Node* original_observer_node) {
 }
 
 static bool outputsNeedToBeObserved(Node* n) {
-  return n->kind() != prim::Constant;
+  return n->kind() != prim::Constant && n->kind() != prim::PythonOp;
 }
 
-void InsertObserverNodes(std::shared_ptr<Graph>& graph, Node* observer_node) {
+void InsertObserverNodes(script::Method* method,
+  std::unordered_map<std::string, Node*>& observerNodeDict) {
+  AT_ASSERT(method != nullptr);
+  auto graph = method->graph();
+  AT_ASSERT(graph != nullptr);
   // For storing all values that need to be instrumented with an observer call.
   std::vector<Value*> values_to_observe;
 
   // For traversing all blocks in the graph including subblocks.
   std::stack<Block*> blocks_to_visit;
 
+  // Observer nodes
+  Node* activation_observer = observerNodeDict.count("activation")
+    ? observerNodeDict["activation"] : nullptr;
```

Comment:
This looks very hacky - again, I'm not convinced that we need an observer for "param" at all.